### PR TITLE
Laundry dryer controls cluster sdk

### PR DESCRIPTION
**Pull request by eahove**
Laundry Dryer controls cluster SDK implemented with the all-clusters-app updated to test this cluster.

Depends on the XML from a previous PR: https://github.com/project-chip/connectedhomeip/pull/30156

### CODE REVIEW

File: src/app/clusters/laundry-dryer-controls-server/laundry-dryer-controls-server.cpp [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+        {
+            DrynessLevelEnum supportedDryness;
+            auto err = delegate->GetSupportedDrynessLevelAtIndex(drynessLevelIdx, supportedDryness);
+            if (err != CHIP_NO_ERROR)
+            {
+                // Can't find the attribute to be written in the supported list (CHIP_ERROR_PROVIDER_LIST_EXHAUSTED)
+                // Or can't get the correct supported list
+                return Status::InvalidInState;
+            }
+            if (supportedDryness == static_cast<DrynessLevelEnum>(*value))
```
Comment (bzbarsky-apple) [inline src/app/clusters/laundry-dryer-controls-server/laundry-dryer-controls-server.cpp:176]: Should statically assert that sizeof(DrynessLevelEnum) == sizeof(*value) or something like that to make this safe.

File: src/app/zap-templates/zcl/zcl-with-test-extensions.json [hunk 2]
```diff
@@ -327,6 +327,7 @@
             "DSTOffsetListMaxSize"
         ],
         "Temperature Control": ["SupportedTemperatureLevels"],
+        "Laundry Dryer Controls": ["SupportedDrynessLevels"],
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/zcl-with-test-extensions.json:330]: Why is this here?  It's a list-typed attribute, right?  This JSON needs to list the non-list-typed attributes that are handled via AttributeAccessInterface.

File: src/app/zap-templates/zcl/zcl.json [hunk 3]
```diff
@@ -325,6 +325,7 @@
             "DSTOffsetListMaxSize"
         ],
         "Temperature Control": ["SupportedTemperatureLevels"],
+        "Laundry Dryer Controls": ["SupportedDrynessLevels"],
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/zcl.json:328]: See above.

**Review by bzbarsky-apple — APPROVED** [review]

File: src/app/zap-templates/zcl/zcl-with-test-extensions.json (hunk 2 above)
Comment (OmAmbalkar) [reply src/app/zap-templates/zcl/zcl-with-test-extensions.json]: Okay, thanks for that, I have removed it in the next commit.

File: src/app/clusters/laundry-dryer-controls-server/laundry-dryer-controls-server.cpp (hunk 1 above)
Comment (OmAmbalkar) [reply src/app/clusters/laundry-dryer-controls-server/laundry-dryer-controls-server.cpp:177]: Added the static assert for the DrynessLevelEnum

**Review by yunhanw-google — APPROVED** [review]
